### PR TITLE
fix more deprecations on v0.5

### PR DESCRIPTION
**Pull request by tlnagy**

### CODE REVIEW

File: src/Gadfly.jl [hunk 1]
```diff
@@ -1023,14 +1023,14 @@ end
...
+    if Compat.is_apple()
         run(`open $(filename)`)
-    elseif OS_NAME == :Linux || OS_NAME == :FreeBSD
+    elseif Compat.is_linux() || Compat.is_bsd()
         run(`xdg-open $(filename)`)
-    elseif OS_NAME == :Windows
+    elseif Compat.is_windows()
         run(`$(ENV["COMSPEC"]) /c start $(filename)`)
     else
-        warn("Showing plots is not supported on OS $(string(OS_NAME))")
```
Comment (yuyichao) [inline src/Gadfly.jl]: No `Compat.` except this one.

Comment (tlnagy) [inline src/Gadfly.jl]: ?

Comment (yuyichao) [inline src/Gadfly.jl]: e.g. `is_windows()` instead of `Compat.is_windows()`

Comment (tlnagy) [inline src/Gadfly.jl]: Ah. Good point.

File: src/Gadfly.jl [hunk 2]
```diff
@@ -1073,10 +1073,10 @@ function display(d::REPLDisplay, ::MIME"text/html", p::Plot)
           </head>
             <body>
             <script charset="utf-8">
-                $(readall(Compose.snapsvgjs))
+                $(Compat.readstring(Compose.snapsvgjs))
```
Comment (yuyichao) [inline src/Gadfly.jl]: Note that my comment applies to these too.

**Comment by tlnagy** [comment]
@yuyichao I went ahead and squashed my commits.

**Pull request merged**